### PR TITLE
Geant4: new recipe

### PR DESCRIPTION
**Pull request by tschoonj**

### CODE REVIEW

**Comment by tschoonj** [comment]
FYI @oliwenman

File: recipes/geant4/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+build:
+    number: '0'
+requirements:
+    build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+    host:
+        - cmake
+        - expat
+        - make  # [unix]
```
Comment (chrisburr) [inline recipes/geant4/meta.yaml]: `make` and `cmake` should be in the `build` requirements section

**Comment by chrisburr** [comment]
I started writing a recipe for this but didn't find time to finish it. It might be useful to copy some parts over: https://github.com/conda-forge/staged-recipes/pull/7807/files

**Comment by tschoonj** [comment]
The macOS build seems to think we're on Linux...

**Comment by tschoonj** [comment]
Oh.. thats because of the cross-linux.cmake file... Fixing

**Comment by tschoonj** [comment]
Is there anything I can do about the macOS build timing out on Travis?

**Comment by chrisburr** [comment]
The feedstock will deploy using Azure so you can ignore it.

File: recipes/geant4/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+package:
+    name: geant4
+    version: {{ version }}
+source:
+    url: http://cern.ch/geant4-data/releases/geant4.10.05.tar.gz
+    sha256: 2a86499d8327abc68456e5d7fc0303824e5704322291b331857cf4042286656e
+build:
+    number: '0'
+    skip: True  # [win]
```
Comment (chrisburr) [inline recipes/geant4/meta.yaml]: You have a `bld.bat` but are skipping Windows, is this intentional?

Comment (tschoonj) [reply recipes/geant4/meta.yaml]: I will remove it and leave the Windows build for someone else to add afterwards.

I am currently writing a test and will update the PR later.

**Comment by tschoonj** [comment]
My test is failing because the conda gcc compilers are not added to the test environment despite having the following `test` section:

```yaml
test:
    requirements:
        - cmake
        - make  # [unix]
        - {{ compiler('c') }}
        - {{ compiler('cxx') }}

```

Any thoughts on how to fix this?

The error I get is:

```bash
/scratch/tmp/tmp12tb62/geant4-10.5.0-hf484d3e_0.tar.bz2: C1115 Found invalid license "Geant4" in info/index.json
TEST START: /scratch/conda-builds/linux-64/geant4-10.5.0-hf484d3e_0.tar.bz2
Renaming work directory,  /scratch/conda-builds/geant4_1553859104415/work  to  /scratch/conda-builds/geant4_1553859104415/work_moved_geant4-10.5.0-hf484d3e_0_linux-64
Collecting package metadata: ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /scratch/conda-builds/geant4_1553859104415/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl


The following NEW packages will be INSTALLED:

    expat:        2.2.5-hf484d3e_1002 conda-forge
    geant4:       10.5.0-hf484d3e_0   local      
    icu:          58.2-hf484d3e_1000  conda-forge
    libgcc-ng:    8.2.0-hdf63c60_1               
    libstdcxx-ng: 8.2.0-hdf63c60_1               
    xerces-c:     3.2.2-hac72e42_1001 conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
WARNING: symlink_conda() is deprecated.
export SRC_DIR=/scratch/conda-builds/geant4_1553859104415/test_tmp
export PREFIX=/scratch/conda-builds/geant4_1553859104415/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: $SRC_DIR
Scanning dependencies of target exampleB1
[ 12%] Building CXX object CMakeFiles/exampleB1.dir/exampleB1.cc.o
c++: error: unrecognized command line option ‘-std=c++17’
c++: error: unrecognized command line option ‘-fno-plt’
make[2]: *** [CMakeFiles/exampleB1.dir/exampleB1.cc.o] Error 1
make[1]: *** [CMakeFiles/exampleB1.dir/all] Error 2
make: *** [all] Error 2
Tests failed for geant4-10.5.0-hf484d3e_0.tar.bz2 - moving package to /scratch/conda-builds/broken
WARNING:conda_build.build:Tests failed for geant4-10.5.0-hf484d3e_0.tar.bz2 - moving package to /scratch/conda-builds/broken
TESTS FAILED: geant4-10.5.0-hf484d3e_0.tar.bz2
```

**Comment by chrisburr** [comment]
I have a couple of ideas, can you push what you have?

**Comment by marcelotrevisani** [comment]
For some reason, your test environment is using ``gcc 4.8``, which is an old compiler

How do you execute the compilation in the test phase?

File: recipes/geant4/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        - cmake
+        - make  # [unix]
+    host:
+        - expat
+        - xerces-c
+    run:
+        - expat
+        - xerces-c
+test:
+    requirements:
```
Comment (marcelotrevisani) [inline recipes/geant4/meta.yaml]: Please, change it to ``requires``

Comment (tschoonj) [reply recipes/geant4/meta.yaml]: In both cases or just in `test`?

Comment (tschoonj) [reply recipes/geant4/meta.yaml]: sorry, reread the docs...

File: recipes/geant4/run_test.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+cmake ${PREFIX}/share/Geant4-${PKG_VERSION}/examples/basic/B1
+make
+source ${PREFIX}/bin/geant4.sh
```
Comment (chrisburr) [inline recipes/geant4/run_test.sh]: The contents of `geant4.(c)sh` should be moved into a post setup hook as is done for ROOT (see [here](https://github.com/conda-forge/root-feedstock/blob/980d1fe10dfd4f365846b9ccc44295d5757f3a99/recipe/build.sh#L147-L156)) and the setup scripts should then be removed as they do harmful things like set `LD_LIBARY_PATH`.

File: recipes/geant4/cross-linux.cmake [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# this one is important
```
Comment (chrisburr) [inline recipes/geant4/cross-linux.cmake]: Is this file actually needed?

File: recipes/geant4/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{% set version = "10.5.0" %}
+
+package:
+    name: geant4
+    version: {{ version }}
+source:
+    url: http://cern.ch/geant4-data/releases/geant4.10.05.tar.gz
```
Comment (chrisburr) [inline recipes/geant4/meta.yaml]: Using something like this will make it easy to update without risking that the version number and sources get out of sync:

```yaml
{% set name = "geant4" %}
{% set version = "10.05" %}
{% set patch_version = "00" %}

package:
  name: {{ name|lower }}
  version: {{ version }}.{{ patch_version }}

source:
{%- if patch_version == "00" %}
  url: http://geant4-data.web.cern.ch/geant4-data/releases/{{ name }}.{{ version }}.tar.gz
{%- else %}
  url: http://geant4-data.web.cern.ch/geant4-data/releases/{{ name }}.{{ version }}.p{{ patch_version }}.tar.gz
{%- endif %}
  sha256: 2a86499d8327abc68456e5d7fc0303824e5704322291b331857cf4042286656e
```

File: recipes/geant4/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    sha256: 2a86499d8327abc68456e5d7fc0303824e5704322291b331857cf4042286656e
+build:
+    number: '0'
```
Comment (chrisburr) [inline recipes/geant4/meta.yaml]:
```suggestion
    number: 0
```

File: recipes/geant4/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+about:
+    summary: 'Geant4 is a toolkit for the simulation of the passage of particles through matter.'
+    dev_url: https://github.com/Geant4/geant4
+    doc_url: https://geant4.web.cern.ch/support
+    home: https://geant4.web.cern.ch/
+    license: Geant4
+    license_file: LICENSE
+extra:
+    recipe-maintainers:
+        - tschoonj
```
Comment (chrisburr) [inline recipes/geant4/meta.yaml]: Do you mind if I am also listed as a maintainer of this?
```suggestion
        - tschoonj
        - chrisburr
```

File: recipes/geant4/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        - xerces-c
+test:
+    requires:
+        - cmake
+        - make  # [unix]
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+about:
+    summary: 'Geant4 is a toolkit for the simulation of the passage of particles through matter.'
+    dev_url: https://github.com/Geant4/geant4
```
Comment (chrisburr) [inline recipes/geant4/meta.yaml]: As I already have this from the old GEANT4 PR:
```suggestion
    dev_url: https://github.com/Geant4/geant4
    description: |
        Geant4 is a toolkit for the simulation of the passage of particles through
        matter. Its areas of application include high energy, nuclear and
        accelerator physics, as well as studies in medical and space science. The
        three main reference papers for Geant4 are published in Nuclear Instruments
        and Methods in Physics Research [A 506 (2003) 250-303](http://www.sciencedirect.com/science/article/pii/S0168900203013688),
        IEEE Transactions on Nuclear Science [53 No. 1 (2006) 270-278](https://ieeexplore.ieee.org/xpls/abs_all.jsp?isnumber=33833&amp;arnumber=1610988&amp;count=33&amp;index=7)
        and Nuclear Instruments and Methods in Physics Research [A 835 (2016) 186-225](http://www.sciencedirect.com/science/article/pii/S0168900216306957).
```

File: recipes/geant4/build.sh [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+else
+  CMAKE_BUILD_TYPE=Release
+fi
+
+mkdir geant4-build
+cd geant4-build
+
+cmake                                                          \
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}                   \
+      -DCMAKE_INSTALL_PREFIX=${PREFIX}                         \
```
Comment (chrisburr) [inline recipes/geant4/build.sh]: I think these options should also be set:
```suggestion
      -DCMAKE_INSTALL_PREFIX=${PREFIX}                         \
      -DGEANT4_BUILD_CXXSTD=17                                 \
      -DGEANT4_USE_SYSTEM_EXPAT=ON                             \
      -DGEANT4_USE_SYSTEM_ZLIB=ON                              \
```
Also what do you think about enabling the graphical components?

File: recipes/geant4/build.sh [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+      -DBUILD_SHARED_LIBS=ON                                   \
+      -DGEANT4_INSTALL_EXAMPLES=ON                             \
+      -DGEANT4_INSTALL_DATA=ON                                 \
```
Comment (chrisburr) [inline recipes/geant4/build.sh]: Would it make more sense to have the data as a separate package?

File: recipes/geant4/cross-linux.cmake (hunk 5 above)
Comment (tschoonj) [reply recipes/geant4/cross-linux.cmake]: According to the [conda docs](https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#an-aside-on-cmake-and-sysroots), yes.

File: recipes/geant4/meta.yaml (hunk 6 above)
Comment (tschoonj) [reply recipes/geant4/meta.yaml]: Not a fan of this as the version will then contain the leading zero in the minor version number, and I need the version number in its current form in `run_test.sh`.

File: recipes/geant4/build.sh (hunk 11 above)
Comment (tschoonj) [reply recipes/geant4/build.sh]: I don't think so.

File: recipes/geant4/build.sh (hunk 10 above)
Comment (tschoonj) [reply recipes/geant4/build.sh]: I would rather keep this recipe _lite_, as Qt drags in half of Anaconda :smile:

File: recipes/geant4/run_test.sh (hunk 4 above)
Comment (tschoonj) [reply recipes/geant4/run_test.sh]: Working on it

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: Actually these scripts are merely setting some environment variables with the locations of the Geant4 data files. Nothing is done to `LD_LIBRARY_PATH` or `PATH` AFAICS.

I could move the contents to the activate/deactivate files but that may cause confusion among the Geant4 users that expect `geant4.(c)sh` to be present.

File: recipes/geant4/run_test.sh [hunk 12]
```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
-cmake ${PREFIX}/share/Geant4-${PKG_VERSION}/examples/basic/B1
```
Comment (chrisburr) [inline recipes/geant4/run_test.sh]: I don't think this will fix it, do you mind if I push a patch to this branch later today?

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: Sure, but please wait until this build has finished, just in case it does fix the problem on macOS :smile:

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: So it failed again... not sure why but it feel suspicious to me that the build and test-phase do not appear to use the same packages (default vs conda-forge).

```
The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-h1de35cc_1002        conda-forge
    ca-certificates: 2019.3.9-hecc5488_0        conda-forge
    cctools:         895-h7512d6f_0             conda-forge
    clang:           4.0.1-h662ec87_0           conda-forge
    clang_osx-64:    4.0.1-h1ce6c1d_11          defaults   
    clangxx:         4.0.1-hc9b4283_0           defaults   
    clangxx_osx-64:  4.0.1-h22b1bf0_11          defaults   
    cmake:           3.14.0-hdd2e4aa_0          conda-forge
    compiler-rt:     4.0.1-h5487866_0           conda-forge
    curl:            7.64.1-h22ea746_0          conda-forge
    expat:           2.2.5-h0a44026_1002        conda-forge
    krb5:            1.16.3-hcfa6398_1001       conda-forge
    ld64:            274.2-h7c2db76_0           conda-forge
    libcurl:         7.64.1-h16faf7d_0          conda-forge
    libcxx:          4.0.1-h579ed51_0           defaults   
    libcxxabi:       4.0.1-hebd6815_0           defaults   
    libedit:         3.1.20170329-hcfe32e1_1001 conda-forge
    libssh2:         1.8.2-hcdc9a53_2           conda-forge
    libuv:           1.26.0-h1de35cc_0          conda-forge
    llvm:            4.0.1-hc748206_0           conda-forge
    llvm-lto-tapi:   4.0.1-h6701bc3_0           conda-forge
    make:            4.2.1-h1de35cc_2004        conda-forge
    ncurses:         6.1-h0a44026_1002          conda-forge
    openssl:         1.1.1b-h01d97ff_2          conda-forge
    rhash:           1.3.6-h1de35cc_1001        conda-forge
    tk:              8.6.9-ha441bb4_1001        conda-forge
    xz:              5.2.4-h1de35cc_1001        conda-forge
    zlib:            1.2.11-h1de35cc_1004       conda-forge
```

```
The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-h1de35cc_1002        conda-forge
    ca-certificates: 2019.3.9-hecc5488_0        conda-forge
    cctools:         895-1                      defaults   
    clang:           4.0.1-1                    defaults   
    clang_osx-64:    4.0.1-h1ce6c1d_11          defaults   
    clangxx:         4.0.1-1                    defaults   
    clangxx_osx-64:  4.0.1-h22b1bf0_11          defaults   
    cmake:           3.14.0-hdd2e4aa_0          conda-forge
    compiler-rt:     4.0.1-hcfea43d_1           defaults   
    curl:            7.64.1-h22ea746_0          conda-forge
    expat:           2.2.5-h0a44026_1002        conda-forge
    geant4:          10.5.0-h0437068_0          local      
    icu:             58.2-h0a44026_1000         conda-forge
    krb5:            1.16.3-hcfa6398_1001       conda-forge
    ld64:            274.2-1                    defaults   
    libcurl:         7.64.1-h16faf7d_0          conda-forge
    libcxx:          4.0.1-hcfea43d_1           defaults   
    libcxxabi:       4.0.1-hcfea43d_1           defaults   
    libedit:         3.1.20170329-hcfe32e1_1001 conda-forge
    libssh2:         1.8.2-hcdc9a53_2           conda-forge
    libuv:           1.26.0-h1de35cc_0          conda-forge
    llvm:            4.0.1-1                    defaults   
    llvm-lto-tapi:   4.0.1-1                    defaults   
    make:            4.2.1-h1de35cc_2004        conda-forge
    ncurses:         6.1-h0a44026_1002          conda-forge
    openssl:         1.1.1b-h01d97ff_2          conda-forge
    rhash:           1.3.6-h1de35cc_1001        conda-forge
    tk:              8.6.9-ha441bb4_1001        conda-forge
    xerces-c:        3.2.2-h44e365a_1001        conda-forge
    xz:              5.2.4-h1de35cc_1001        conda-forge
    zlib:            1.2.11-h1de35cc_1004       conda-forge
```

File: recipes/geant4/build.sh [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+if [[ ${DEBUG_C} == yes ]]; then
+  CMAKE_BUILD_TYPE=Debug
+else
+  CMAKE_BUILD_TYPE=Release
+fi
+
+# cmake_minimum_required(VERSION 3.1) is required to compile
+# the examples to compile using conda's compiler packages
+sed -r -i -E 's#cmake_minimum_required\(VERSION [0-9]\.[0-9]#cmake_minimum_required(VERSION 3.1#gI' \
```
Comment (chrisburr) [inline recipes/geant4/build.sh]: @tschoonj So the issue on macOS is that `cmake_minimum_required(VERSION 3.1)` sets a policy in CMake that is required for it to support the conda compilers.

This line patches all the examples so users of the package don't run into the same issue. If you would rather fix it a different way, go ahead.

File: recipes/geant4/run_test.sh (hunk 12 above)
Comment (chrisburr) [reply recipes/geant4/run_test.sh]: That's fine, conda-forge's compiler packages come from anaconda to avoid issues with building the compilers and to improve the interoperability between the channels. Some of them have been manually copied over but these haven't.

File: recipes/geant4/meta.yaml (hunk 6 above)
Comment (chrisburr) [reply recipes/geant4/meta.yaml]: When fixing the testing issue I also edited this to pad the zeros only for the URLs. What do you think of this version?

File: recipes/geant4/run_test.sh (hunk 4 above)
Comment (chrisburr) [reply recipes/geant4/run_test.sh]: Ah, the [LCG](https://ep-dep-sft.web.cern.ch/document/lcg-releases) build I looked at has them differently.

I still find it nicer to have the environment variables setup automatically rather than relying on on users remembering to source `geant4.sh`. It can also unset the variables when changing conda environments. I also added dummy `geant4.(c)sh` scripts to ensure they're not picked up from from a different GEANT4 installation. What do you think?

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: These activation scripts won't be invoked in the root environment right? So there won't be an easy way for the user to get these variables set up then?

I don't think there's anything wrong with the user relying on these scripts since this is what the Geant4 docs tell people to use.

File: recipes/geant4/build.sh (hunk 13 above)
Comment (tschoonj) [reply recipes/geant4/build.sh]: No looks good to me

File: recipes/geant4/meta.yaml (hunk 6 above)
Comment (tschoonj) [reply recipes/geant4/meta.yaml]: That's some nice jinja templating! Neat!

File: recipes/geant4/run_test.sh (hunk 4 above)
Comment (chrisburr) [reply recipes/geant4/run_test.sh]: If a "recent" conda installation is used they will be set automatically .`conda` is now a function sourced from a script (`$CONDA_DIR/etc/profile.d/conda.sh`) rather than requiring you to editing your PATH. This means it doesn't affect anything until you run `conda activate` and `conda install` or `conda activate` can transparently modify environment variables.

> So there won't be an easy way for the user to get these variables set up then?

`conda activate` or `conda activate root` should work on newer installations and `source activate` or `souce activate root` should work on any installation. Some other conda packages won't work without the environment being set up correctly (most notably the compilers) and I've patched the `geant4.(c)sh` scripts to print a message telling users what to do.

Does this seem reasonable? If you disagree, you can remove the last 5 lines of the build script.

I know I'm a little idealistic about this but I really don't think people should ever have to source magic scripts to make packages work correctly, once the environment is active everything should just work. Personally I would prefer to include the default paths inside the binaries and for the environment variables to be used only for overriding them but that's a suggestion for another day...

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: While the decision of the Geant4 devs to expect the user to source a shell script to get the library working is certainly questionable, it is one we have to live with.

Therefore I believe that sourcing the shell script should always result in a working library, and this should be independent of whether the user activated the root environment or not.

I am therefore in favor of keeping the Geant4 scripts as generated by cmake, and dispose of the activation scripts.

Comment (chrisburr) [reply recipes/geant4/run_test.sh]: Why not keep both?

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: I think one of the reasons that the Geant4 developers introduced this shell script is to allow users to override the default datasets.

If you're going to rely on activation scripts to set the environment variables, then things may get messy quickly as the value of these variables will then depend on the order of the scripts that get sourced: the ones provided by the activation script or the script with the custom variables.

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: Hi Chris... any final thoughts on this?

Comment (chrisburr) [reply recipes/geant4/run_test.sh]: Sorry I missed your last reply and I hope I'm not being too awkward about this, I just want GEANT4 to be packaged as nicely as possible as it's a dependency of some other packages I will be adding.

I think it's more likely to get messy if the activation scripts don't set these variables. If someone runs `source geant4.sh` then `conda activate my-environment` they will get these variables from a different GEANT4 installation which probably isn't what they want.

Also I just noticed `geant4.(c)sh` does change the `PATH` and `LD_LIBRARY_PATH`[1] on Linux and this will break things in unexpected ways.

How about some other ideas (which aren't mutually exclusive):

1. The activation script print a warning if any of the variables are already set.
```bash
$ export G4NEUTRONHPDATA=/my/custom/data/directory
$ conda activate my-environment
WARNING: $G4NEUTRONHPDATA has been changed from "/my/custom/data/directory" to "/conda/share/Geant4/data/G4NDL4.5"
$ 
```
2. Create a package called `geant4-base` which doesn't include the data at all
    * There is then metapackage called `geant4` which depends on `geant4-base`,  `geant4-data-g4ndl` and other `geant4-data-.*` packages which are pinned to the default versions for that geant4 release but users can still install different versions of the data if they want.
    * The `$G4.*DATA` variables are then set by each of the data packages.
    * Actually I really like this idea as as there can be four variants of  `geant4`:
        * `default`: No QT support and pinned to the exact data package versions
        * `custom_data`: No QT support and any version of the data packages can be used
        * `qt-default`: QT support and pinned to the exact data package versions
        * `qt-custom_data`: QT support and any version of the data packages can be used
    * Running `conda install geant4-data-g4ndl=4.3` will automatically switch to the `custom_data` variant as version 4.5 is the default.
    * Downstream packages depend on `geant4` which will can be any of the four variants. This is way that `numpy` has support for both `openblas` and `mkl`.

Having thought about it I'm inclined to implement both ideas, what do you think?

[1] Search for "Contents of $PREFIX/bin/geant4.sh is" in [this log](https://circleci.com/gh/conda-forge/staged-recipes/36812?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

Comment (tschoonj) [reply recipes/geant4/run_test.sh]: Sure sounds good but looks like a lot of work.

**Comment by chrisburr** [comment]
The failures can be ignored:
 * Travis: Times out but the feedstock will use Azure so it's not a problem.
 * Windows: I haven't bothered to write a build script for Windows but they're marked as `noarch` so they'll only build on Linux. They do include a Windows activation script for the environment variables.

@tschoonj This should implement my suggestions from https://github.com/conda-forge/staged-recipes/pull/8067#discussion_r272501201.

* Installing only `geant4` will give a installation without QT support with the default set of data package versions.
* Manually specifying a different version of a data package will automatically switch to `geant4_data_variant=*=custom`. (Not currently possible as there is only one version of each but it's there in preparation for future releases.)
* I haven't add the build with `qt` support yet but most of the necessary changes are included so the `# qt` line can be uncommented in the feedstock's `conda_build_config.yaml` file. It will then be installable using:

```bash
$ conda install geant4 'geant4_visualisation_variant=*=qt'
```

Are you happy with these changes and with being listed as a maintainer of all the data packages?

**Comment by tschoonj** [comment]
Nice! Everything looks good to me, as far as I understand what you're doing 😄 

I am ok being a maintainer for all the packages.

Thanks!

**Comment by chrisburr** [comment]
Then LGTM 👍 If you have any questions about how it works or what it's doing feel free to ask.

@colasri As you showed interest in https://github.com/conda-forge/staged-recipes/pull/7908, GEANT4 will be available once this deploys.
